pkg/crypto/keys/publickey.go

```diff
@@ -100,24 +100,34 @@ func NewPublicKeyFromASN1(data []byte) (*PublicKey, error) {
 
 // decodeCompressedY performs decompression of Y coordinate for given X and Y's least significant bit.
 func decodeCompressedY(x *big.Int, ylsb uint) (*big.Int, error) {
-	c := elliptic.P256()
-	cp := c.Params()
-	three := big.NewInt(3)
-	/* y**2 = x**3 + a*x + b  % p */
-	xCubed := new(big.Int).Exp(x, three, cp.P)
-	threeX := new(big.Int).Mul(x, three)
-	threeX.Mod(threeX, cp.P)
-	ySquared := new(big.Int).Sub(xCubed, threeX)
-	ySquared.Add(ySquared, cp.B)
-	ySquared.Mod(ySquared, cp.P)
-	y := new(big.Int).ModSqrt(ySquared, cp.P)
+	cp := elliptic.P256().Params()
+
+	// x ^ 3 mod P
+	x3 := new(big.Int).Exp(x, new(big.Int).SetInt64(3), cp.P)
+
+	// ax mod P
+	ax := new(big.Int).Mul(x, new(big.Int).SetInt64(-3))
+	ax.Mod(ax, cp.P)
+
+	// x^3 + ax mod P
+	x3.Add(x3, ax)
```

Comment:
It is not an `x^3`, it will be an `y^2`. Why have you changed names?

Reply:
as said in comment: x^3 + ax mod P, not an y^2